### PR TITLE
Adding properties description for User Model

### PR DESCRIPTION
**Pull request by davidcheung**
- Added description for `realm` `created` `lastUpdated` `status`

connected to #1115

### CODE REVIEW

**Comment by richardpringle** [comment]
@elkorep, Could you please take a look at the failed checks when you get a chance?

File: common/models/user.js [hunk 1]
```diff
@@ -42,11 +42,15 @@ var debug = require('debug')('loopback:user');
...
- * @property {String} password Hidden from remote clients
- * @property {String} email Must be valid email
- * @property {Boolean} emailVerified Set when a user's email has been verified via `confirm()`
- * @property {String} verificationToken Set when `verify()` is called
+ * @property {String} username Must be unique.
+ * @property {String} password Hidden from remote clients.
+ * @property {String} email Must be valid email.
+ * @property {Boolean} emailVerified Set when a user's email has been verified via `confirm()`.
+ * @property {String} verificationToken Set when `verify()` is called.
+ * @property {String} realm Specific domain a user belongs to, [Learn more](https://docs.strongloop.com/display/public/LB/Partitioning+users+with+realms).
```
Comment (davidcheung) [inline common/models/user.js]: @bajtos  Not too familiar with the realm concept, can you take a look to see if this make sense?

Comment (bajtos) [inline common/models/user.js]: I'd rephrase the sentence as follows.

```
@property {String} realm The namespace the user belongs to. See [Partitioning users with realms](https://docs.strongloop.com/display/public/LB/Partitioning+users+with+realms) for details.
```

@crandmck could you please review?

File: common/models/user.js [hunk 2]
```diff
@@ -42,11 +42,15 @@ var debug = require('debug')('loopback:user');
...
+ * @property {Date} created Unused properties (Deprecated).
```
Comment (bajtos) [inline common/models/user.js]: I am proposing a slightly different text (please apply it to all three unused properties):

```
* @property {Date} created The property is not used by LoopBack, you are free to use it for your own purposes. 
```

On the second thought, I think it's not necessary to mention that we are thinking about removing this property in the future. IMO it's enough to mention the removal once it's done, because the upgrade path would be pretty straightforward (define these three properties in your subclassed User model).

@crandmck please review

**Comment by davidcheung** [comment]
@crandmck can you please review

**Comment by crandmck** [comment]
LGTM

**Comment by crandmck** [comment]
We should be able to merge this, as it's just additions to doc comments, however CI tests are failing.  Unsure if merging will cause a problem?

**Comment by rmg** [comment]
@slnode test please (I fixed failures in some of the dependent modules)

**Comment by davidcheung** [comment]
@slnode test please

**Comment by bajtos** [comment]
I'll go ahead and land this, despite the failing dependants builds. Doc comments should not cause any downstream failures.
